client/containers/Document/Document.js

```diff
@@ -13,148 +15,321 @@ const propTypes = {
-				<div className="container">
-					<div id="flex-container" className="row">
-						<div className="col-12">
-							<header>
-								<h1 className={generatedTitle.isPlaceholder ? 'placeholder' : ''}>
-									{generatedTitle.title}
-								</h1>
-								<section className={description ? '' : 'placeholder'}>
-									{description || placeholderDescription}
-								</section>
-							</header>
-							<main>
-								<div>
-									<section>
-										<HTMLTable {...tableProps}>
-											<tbody>
-												<tr>
-													<td>Source</td>
-													<td>
-														<a
-															href={`/organization/${organizationSlug}`}
-														>
-															{organizationName}
-														</a>
-													</td>
-												</tr>
-												<tr>
-													<td>Created</td>
-													<td>{createdAt}</td>
-												</tr>
-												<tr>
-													<td>Updated</td>
-													<td>{updatedAt}</td>
-												</tr>
-												<tr>
-													<td>Published</td>
-													<td>{publicationDate}</td>
-												</tr>
-												<tr>
-													<td>Language</td>
-													<td>{language}</td>
-												</tr>
-												<tr>
-													<td>Content Type</td>
-													<td>{contentType}</td>
-												</tr>
-												<tr>
-													<td>CPC Codes</td>
-													<td>
-														<ul>
-															{cpcCodes.map((code) => (
-																<li key={code}>{code}</li>
-															))}
-														</ul>
-													</td>
-												</tr>
-											</tbody>
-										</HTMLTable>
-									</section>
-									<section>
-										<h2>Files</h2>
-										{props.assertionData.map((assertion) => (
-											<HTMLTable key={assertion.cid} {...tableProps}>
+class Document extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			documentData: {
+				...props.documentData,
+			},
+			documentNewData: pick(props.documentData, ['id', 'title', 'description']),
+			hasChanged: false,
+			putDocumentIsLoading: false,
+			putDocumentError: undefined,
+		};
+		this.onTitleChange = this.onTitleChange.bind(this);
+		this.onDescriptionChange = this.onDescriptionChange.bind(this);
+		this.handlePutDocument = this.handlePutDocument.bind(this);
+		this.handleEditDocumentDataClick = this.handleEditDocumentDataClick.bind(this);
+		this.handleCancelDocumentDataEditClick = this.handleCancelDocumentDataEditClick.bind(this);
+	}
+
+	onTitleChange(evt) {
+		const { documentNewData } = this.state;
+		const documentNewTitle = evt.target.value;
+		if (documentNewData.title !== documentNewTitle) {
+			this.setState((prevState) => ({
+				documentNewData: {
+					...prevState.documentNewData,
+					title: documentNewTitle,
+				},
+				hasChanged: true,
+			}));
+		}
+	}
+
+	onDescriptionChange(evt) {
+		const { documentNewData } = this.state;
+		const documentNewDescription = evt.target.value;
```

Comment:
`newDescription` seems good!

Reply:
What do you think of applying that pattern to the other instances of "document" (e.g. newData)

I think that fixes the legibility concerns all around.

Reply:
Well, we're talking two different contexts.

`newDescription` is a local constant within the `onDescriptionChange` function. Its context is clear. `documentNewData` is a state key. I prefer it to be highly explicit there.

If you think we should reassign it when destructuring, so the function-local name is `newData`, that's… fine. I won't object. But I'd hate to lose the specificity when talking about construction of the state object.

That said this all needs refactoring anyway yes.

Reply:
…although maybe since we're talking about the sate for the `DocumentContainer`, `document` is a totally unnecessary prefix anyway.

HEY ONCE WE FIGURE THIS OUT LET'S TACKLE CACHING